File: android/src/main/java/com/xgs/flutter_live/BJYLiveOption.java
package com.xgs.flutter_live;

import com.google.gson.Gson;

import javax.annotation.Nonnull;

import io.flutter.plugin.common.MethodCall;

/**
 * Created  on 2019/10/26.
 *
 * @author grey
 */
class BJYLiveOption {

    private String userName;
    private String avatarUrl;
    private String userNum;
    private String sign;
    private String roomId;
    ///大班课
    private boolean interactive;

    BJYLiveOption create(@Nonnull MethodCall call) {
        this.userName = call.argument("userName");
        this.userNum = call.argument("userNum");
        this.avatarUrl = call.argument("userAvatar");
        this.sign = call.argument("sign");
        this.roomId = call.argument("roomId");
        this.interactive = call.argument("interactive");
        return this;
    }

    String getUserName() {
        return userName == null ? "" : userName;
    }

    String getAvatarUrl() {
        return avatarUrl == null ? "" : avatarUrl;
    }

    String getUserNum() {
        return userNum == null ? "" : userNum;
    }

    String getSign() {
        return sign == null ? "" : sign;
    }

    public boolean isInteractive() {
        return interactive;
    }

    Long getRoomId() {
        try {
            return Long.parseLong(roomId);
        } catch (Exception e) {
            return 0L;
        }
    }


}

File: android/src/main/kotlin/com/xgs/flutter_live/BJYLiveOption.kt
package com.xgs.flutter_live

import io.flutter.plugin.common.MethodCall
import javax.annotation.Nonnull

/**
 * Created  on 2019/10/26.
 *
 * @author grey
 */
class BJYLiveOption {
    private var userName: String? = null
    private var avatarUrl: String? = null
    private var userNum: String? = null
    private var sign: String? = null
    private var roomId: String? = null
    fun create(@Nonnull call: MethodCall): BJYLiveOption {
        userName = call.argument("userName")
        userNum = call.argument("userNum")
        avatarUrl = call.argument("userAvatar")
        sign = call.argument("sign")
        roomId = call.argument("roomId")
        return this
    }

    fun getUserName(): String {
        return if (userName == null) "" else userName!!
    }

    fun getAvatarUrl(): String {
        return if (avatarUrl == null) "" else avatarUrl!!
    }

    fun getUserNum(): String {
        return if (userNum == null) "" else userNum!!
    }

    fun getSign(): String {
        return if (sign == null) "" else sign!!
    }

    fun getRoomId(): Long {
        return try {
            roomId?.toLong()?:0L
        } catch (e: Exception) {
            0L
        }
    }
}